src/libraries/System.Data.Common/src/System/Data/Common/DbProviderFactory.cs

```diff
@@ -35,28 +42,33 @@ public virtual bool CanCreateCommandBuilder
             {
                 if (!_canCreateCommandBuilder.HasValue)
                 {
-                    using (DbCommandBuilder builder = CreateCommandBuilder())
+                    try
+                    {
+                        using var adapter = CreateCommandBuilder();
+                        _canCreateCommandBuilder = true;
+                    }
+                    catch
                     {
-                        _canCreateCommandBuilder = builder != null;
+                        _canCreateCommandBuilder = false;
                     }
                 }
 
                 return _canCreateCommandBuilder.Value;
             }
         }
 
-        public virtual DbCommand CreateCommand() => null;
+        public virtual DbCommand CreateCommand() => throw new NotSupportedException();
```

Comment:
Note that this is where I'm proposing a behavioral breaking change (see comments in the PR text body).

Reply:
Discussion continues below: https://github.com/dotnet/runtime/pull/689#issuecomment-650124113